### ISSUE TITLE
preserve existing testing workflow sans tox

### ISSUE THREAD

**Issue by kapilt** [capitalone/cloud-custodian#1324]
tox maybe okay for ci, but it wastes way too much time and spews too much junk noise vs the previous `make test` setup.

**Comment by kapilt** [comment]
hmm the line noise seems to be an artifact of my working copy and non explicit directories given to flake8 tox step

**Comment by kapilt** [comment]
ratchet.py should also be in tools/dev

**Comment by kapilt** [comment]
deep dived and found a bunch of issues, std overhead on tox seems to be about 10s, but some tweaks to config and test env, drops runtime costs by over a minute.

**Comment by kapilt** [comment]
running nosetests with parallel gets to 642 tests in 23s. 

okay, couple of items

- [ ] add make test target back alias to tox -e py27
- [ ] drop rachet shell front end use tox config for ignore exit code
- [ ] support pypy env in tox
- [ ] default to showing 30 slowest tests in ci
- [ ] fix ebs snapshot timeout (30s slow down)
- [ ] 3 s3 tests fail with nosetests in parallel
- [ ] leaking tmp directories


speed check with changes / py.test
```
$ time py.test -n auto tests
======================================== test session starts ========================================
platform darwin -- Python 2.7.13, pytest-3.1.2, py-1.4.34, pluggy-0.4.0
rootdir: /Users/ylv522/projects/custodian, inifile: tox.ini
plugins: xdist-1.17.1, cov-2.5.1
gw0 [642] / gw1 [642] / gw2 [642] / gw3 [642] / gw4 [642] / gw5 [642] / gw6 [642] / gw7 [642]
scheduling tests via LoadScheduling
.............................................................................................................................................................................................................................................................................................\
.................................................................................................................................................................................................................................................................s...........................\
........................................................................
============================== 641 passed, 1 skipped in 24.33 seconds ===============================

real    0m24.765s
user    2m8.562s
sys     0m13.531s
```

speed check for nosetests shows some errors which is concerning

```
$ time nosetests -s tests --processes=-1
.............................................................................................................................................................................................................................................................................................\
.............................................................................................................................................................................................................................................................................................\
.............................................Exception nose.plugins.multiprocess.TimedOutException: TimedOutException() in <function _removeHandlerRef at 0x107e94de8> ignored
S....................EEE.E.
Ran 642 tests in 22.894s
```

tests with failures
 - TimedOutException: 'test_s3_remove_tag (tests.test_s3.S3Test)'
 - TimedOutException: 'test_remove_policy_none_extant (tests.test_s3.S3Test)'
 - TimedOutException: 'test_remove_policy (tests.test_s3.S3Test)'
 - TimedOutException: 'test_no_encryption_statement (tests.test_s3.S3Test)'

**Comment by kapilt** [comment]
closed in #1326 the nose tests parallel still need some investigation, but make test works, and tests are significantly faster ( at least 2x in ci), also output is less voluminous.
